core/Files.cc

```diff
@@ -114,12 +114,8 @@ File::File(string &&path_, string &&source_, Type sourceType, uint32_t epoch)
     : epoch(epoch), sourceType(sourceType), flags(path_), packagedLevel{File::filePackagedSigil(source_)},
       path_(move(path_)), source_(move(source_)), originalSigil(fileStrictSigil(this->source_)),
       strictLevel(originalSigil) {
-    if (this->source_.size() > INT32_MAX) [[unlikely]] {
-        // Specifically looking for INT32_MAX, because File::lineBreaks_ stores `int`, not `uint32_t`
-        static_assert(INT32_MAX < INVALID_POS_LOC);
-        static_assert(is_same_v<decltype(LocOffsets::beginLoc), uint32_t>);
-        static_assert(is_same_v<remove_reference_t<decltype((*this->lineBreaks_)[0])>, int>);
-        Exception::raise("File larger than {} bytes. Got: {}", INT32_MAX, this->source_.size());
+    if (this->source_.size() >= INVALID_POS_LOC) [[unlikely]] {
+        Exception::raise("File not less than {} bytes. Got: {}", UINT32_MAX, this->source_.size());
```

Comment:
This double's Sorbet's max supported file size, expanding to files with size `2^32-1`, instead of `2^31`.